### PR TITLE
`1es-template`-ize partner release pipeline

### PR DESCRIPTION
**Pull request by scbedd**
[Test Release](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3615890&view=results)

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: eng/pipelines/templates/stages/1es-redirect.yml [hunk 1]
```diff
@@ -39,7 +39,7 @@ extends:
       binskim:
         # Exclude imported azure-sdk-build-tools gpg/azcopy binaries
         # See https://dev.azure.com/securitytools/SecurityIntegration/_wiki/wikis/Guardian/1378/Glob-Format
-        analyzeTargetGlob: +:file|**/*.jar;+:file|**\*.jar;+:file|**/*.exe;-:f|**/tools/gpg/**/*.dll;-:f|**/tools/gpg/**/*.exe;-:f|**/tools/azcopy/**/*.exe;-:f|**/tools/azcopy/**/*.dll
+        analyzeTargetGlob: +:file|**/*.jar;+:file|**/*.exe;-:f|**/tools/gpg/**/*.dll;-:f|**/tools/gpg/**/*.exe;-:f|**/tools/azcopy/**/*.exe;-:f|**/tools/azcopy/**/*.dll
```
Comment (benbp) [inline eng/pipelines/templates/stages/1es-redirect.yml]: Did the issue end up not being `\`?

Comment (scbedd) [reply eng/pipelines/templates/stages/1es-redirect.yml]: Nope. Backed that out.

File: eng/pipelines/partner-release.yml [hunk 2]
```diff
@@ -11,73 +11,80 @@ parameters:
...
+    stages:
+      - stage:
+        displayName: 'Partner Release'
+        variables:
+          - name: BuildToolScripts
+            value: $(Pipeline.Workspace)/azure-sdk-build-tools/scripts
+          - name: Artifacts
+            value: $(Pipeline.Workspace)/artifacts
+          - name: EsrpArtifacts
+            value: $(Build.SourcesDirectory)/packages-esrp
```
Comment (scbedd) [inline eng/pipelines/partner-release.yml:26]: This changed from `$(Pipeline.Workspace)/packages-esrp` -> `$(Build.SourcesDirectory)/packages-esrp`.

I tried everything, including copying the downloaded unsigned artifacts INTO the `packages-esrp` as a subfolder. Still doesn't scan it properly.

The instant I moved it under `$(Build.SourcesDirectory)`, binskim errors disappeared.

**Comment by scbedd** [comment]
/check-enforcer override

The failing tests are unrelated to this PR, which only affects the partner release pipeline.

**Pull request merged**